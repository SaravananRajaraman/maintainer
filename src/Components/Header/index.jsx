import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import styled from 'styled-components';

const StyledHeader = styled(AppBar)`    
        background-image: linear-gradient(#314755 0%, #26a0da 51%, #314755 100%);         
        align-items : center;   
    `   

export default function Header() {  

  return (    
      <StyledHeader data-test='Header-Component' position="static">
        <Toolbar variant="dense">                     
            <Typography data-test='Title' variant="h6" color="inherit">
                Maintainers Info
            </Typography>
        </Toolbar>
      </StyledHeader>    
  );
}
