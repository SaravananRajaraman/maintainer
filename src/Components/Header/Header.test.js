import React from 'react';
import {shallow} from 'enzyme';
import {findTestAttr} from './../../Utils';
import Header from './index';
import toJson from 'enzyme-to-json';

const setUp =()=>{
    const component = shallow(<Header/>);
    return component;
}

describe('Header Component', ()=>{

    let component;
    beforeEach(()=>{
        component = setUp();
    })

    it('Should render the component with no error', ()=>{        
        let wrapper = findTestAttr(component,'Header-Component');
        expect(wrapper.length).toBe(1);
    });

    it('Should render the Title', ()=>{        
        let wrapper = findTestAttr(component,'Title');
        expect(wrapper.length).toBe(1);
    });   

    it('Should match the snapShot', ()=>{                
        expect(toJson(component)).toMatchSnapshot();
    });

});
