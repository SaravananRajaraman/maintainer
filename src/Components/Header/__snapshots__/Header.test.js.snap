// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should match the snapShot 1`] = `
<Styled(WithStyles(ForwardRef(AppBar)))
  data-test="Header-Component"
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))
    variant="dense"
  >
    <WithStyles(ForwardRef(Typography))
      color="inherit"
      data-test="Title"
      variant="h6"
    >
      Maintainers Info
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Toolbar))>
</Styled(WithStyles(ForwardRef(AppBar)))>
`;
